pwndbg/commands/__init__.py

```diff
@@ -14,19 +14,24 @@
 from typing import Tuple
 from typing import TypeVar
 
-import gdb
 from typing_extensions import ParamSpec
 
 import pwndbg.exception
-import pwndbg.gdblib.heap
-import pwndbg.gdblib.kernel
-import pwndbg.gdblib.proc
-import pwndbg.gdblib.qemu
-import pwndbg.gdblib.regs
-from pwndbg.gdblib.heap.ptmalloc import DebugSymsHeap
-from pwndbg.gdblib.heap.ptmalloc import GlibcMemoryAllocator
-from pwndbg.gdblib.heap.ptmalloc import HeuristicHeap
-from pwndbg.gdblib.heap.ptmalloc import SymbolUnresolvableError
+
+# These aren't available under LLDB, and we can't get rid of them until all of
+# this functionality has been ported to the Debugger API.
+#
+# TODO: Replace these with uses of the Debugger API.
```

Comment:
What does it mean that it has to be replaced with Debugger API?

Is there going to be a method for applying debugger-specific imports or registering commands?

Reply:
> What does it mean that it has to be replaced with Debugger API?

`OnlyWithResolvedHeapSyms` need these bits to work. So, if we want access to it under LLDB, we need to port them to the debugger-agnostic APIs. The plan is to replace these imports with the generic imports once they're all available.

> Is there going to be a method for applying debugger-specific imports or registering commands?

I'm not sure what you mean by debugger-specific imports besides what we're already doing with the `pwndbg.dbg.is_gdblib_available()` checks. If it makes sense for us to distinguish between more than just GDB v. everything else, that should be easy enough to add, but there are no LLDB specific imports as of the latest changes in the port branch.

Debugger-specific command registration can be done by just importing debugger-specific commands in the debugger's `Debugger.setup()` method. `pset` under LLDB does just that. I don't expect there to be enough of those that it warrants a separate mechanism.